<?xml version="1.0"?>
<launch>
  <!-- Arma 3 Depth Image Receiver -->
  
  <!-- TCP 服务器端口 -->
  <arg name="tcp_port" default="5555"/>
  
  <!-- ROS 话题名称 -->
  <arg name="depth_topic" default="/camera/depth/image_rect_raw"/>
  <arg name="camera_info_topic" default="/camera/depth/camera_info"/>
  
  <!-- 相机内参（与 EGO-Planner 的 camera.yaml 一致） -->
  <arg name="cam_width" default="640"/>
  <arg name="cam_height" default="480"/>
  <arg name="cam_fx" default="386.02557373046875"/>
  <arg name="cam_fy" default="386.02557373046875"/>
  <arg name="cam_cx" default="321.8554382324219"/>
  <arg name="cam_cy" default="241.2396240234375"/>
  
  <!-- 启动深度图像接收节点 -->
  <node pkg="arma3_ros_bridge" type="arma3_depth_receiver.py" name="arma3_depth_receiver" output="screen">
    <param name="tcp_port" value="$(arg tcp_port)"/>
    <param name="depth_topic" value="$(arg depth_topic)"/>
    <param name="camera_info_topic" value="$(arg camera_info_topic)"/>
    <param name="cam_width" value="$(arg cam_width)"/>
    <param name="cam_height" value="$(arg cam_height)"/>
    <param name="cam_fx" value="$(arg cam_fx)"/>
    <param name="cam_fy" value="$(arg cam_fy)"/>
    <param name="cam_cx" value="$(arg cam_cx)"/>
    <param name="cam_cy" value="$(arg cam_cy)"/>
  </node>
  
</launch>
